src/customisations/components/views/messages/ContentScanningDownloadActionButton.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MatrixEvent } from "matrix-js-sdk/src/matrix";
+import React from "react";
+import classNames from "classnames";
+import { MediaEventHelper } from "matrix-react-sdk/src/utils/MediaEventHelper";
+import { FileDownloader } from "matrix-react-sdk/src/utils/FileDownloader";
+import { _t, _td } from "matrix-react-sdk/src/languageHandler";
+import Spinner from "matrix-react-sdk/src/components/views/elements/Spinner";
+import { RovingAccessibleTooltipButton } from "matrix-react-sdk/src/accessibility/RovingTabIndex";
+
+import { Media } from "../../../ContentScanningMedia";
+import { BlockedIcon } from "../../../../components/views/elements/BlockedIcon";
+
+interface IProps {
+    mxEvent: MatrixEvent;
+
+    // XXX: It can take a cycle or two for the MessageActionBar to have all the props/setup
+    // required to get us a MediaEventHelper, so we use a getter function instead to prod for
+    // one.
+    mediaEventHelperGet: () => MediaEventHelper;
+}
+
+interface IState {
+    loading: boolean;
+    blob?: Blob;
+    tooltip: string;
+    isSafe: boolean;
+    isScanning: boolean;
+    hasError: boolean;
+    downloadClicked: boolean;
+}
+
+/**
+ * Replacement for the default download action button.
+ * Displays content scanning status and prohibits downloading unsafe content.
+ */
+export default class ContentScanningDownloadActionButton extends React.PureComponent<IProps, IState> {
+    private downloader = new FileDownloader();
+
+    public constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            hasError: false,
+            isSafe: false,
+            isScanning: false,
+            loading: false,
+            tooltip: _td("Downloading"),
```

Comment:
not sure this is the right init value. "Download" would make more sense, or just leave `render` to figure it out ?

Reply:
This state property can be removed anyway.